### ISSUE TITLE
Hair and Face Collision

### ISSUE THREAD

**Issue by sunshineatnoon** [c-he/perm#13]
Hi, Thanks for releasing the code. I sampled some random hair style from the pre-trained model and noticed that in many cases the hair and face are colliding (i.e., hair occludes large portion of the face). I don't see these collision in the paper. Is it normal?

![Image](https://github.com/user-attachments/assets/2fa6dc7e-fc18-44cb-a0ef-18d7ad5dde1c)
![Image](https://github.com/user-attachments/assets/df7e69ad-9717-497c-a1a6-b24d88460e8f)
![Image](https://github.com/user-attachments/assets/c44fb369-b9e0-4add-ba05-1f7bf6c82f4c)

**Comment by c-he** [comment]
I would say it is normal because the first two samples just contain long bangs that cover the upper face (imagine some「杀马特」hairstyles) and the last sample shows some annoying penetrated strands. We did encountered similar problems and for the figures shown in our paper, we typically ran some collision resolving algorithms first ro remove these annoying penetrations. Because the collision resolving code is from our internal codebase, I could not share it, but the algorithm we used to implement it is called Follow-the-Leader (FTL) algorithm.

**Comment by sunshineatnoon** [comment]
Thanks for the response. I am wondering if this happens because of the training data?

**Comment by c-he** [comment]
Yes training data is part of the reason. You can also try to add collision loss similar to https://arxiv.org/pdf/1806.07467 to suppress these artifacts.

**Comment by sunshineatnoon** [comment]
Thanks for the info!
